pandas/core/arrays/datetimes.py

```diff
@@ -1574,6 +1643,24 @@ def sequence_to_dt64ns(data, dtype=None, copy=False,
     # if dtype has an embedded tz, capture it
     tz = validate_tz_from_dtype(dtype, tz)
 
+    if isinstance(data, ABCIndexClass):
+        data = data._data
+
+    if isinstance(data, DatetimeArrayMixin):
+        # series / index have been unboxed. If we're here, we just
+        # need to validate against user-provided parameters and exit early.
+        if tz and data.tz:
+            if not timezones.tz_compare(tz, data.tz):
+                msg = (
+                    "Timezone of the array and 'dtype' do not match. "
+                    "'{}' != '{}'"
+                )
+                raise TypeError(msg.format(tz, data.tz))
+        tz = data.tz
+        tz = validate_tz_from_dtype(dtype, tz)
+
+        return data._data, tz, data.freq
+
```

Comment:
I think remove this chunk and instead wrap L1685 in np.asarray, then you’ll consistently get ndarray back

Reply:
they've already been extracted, should be retained fine

Reply:
What line extracts the tz? I don't see one.

Reply:
FYI, this does return an ndarray.

Reply:
> What line extracts the tz? I don't see one.

`tz = maybe_infer_tz(tz, data.tz)`